server/controllers/api.py

```diff
@@ -606,25 +606,34 @@ def get(self, user, name):
                 return restful.abort(404)
             return restful.abort(403)
 
+        limit = request.args.get('limit', 150, type=int)
+        offset = request.args.get('offset', 0, type=int)
+
         if not self.model.can(assign, user, 'export'):
             return restful.abort(403)
 
-        students, subms, no_subms = assign.course_submissions()
+        subms = assign.course_submissions(include_empty=False)
 
         output = []
-        subm_keys = sorted(list(subms))
-
-        for s_id in subm_keys:
-            output.append(models.Backup.query.get(s_id))
-
-        num_subms = len(output)
+        subm_keys = set(s['backup']['id'] for s in subms)
+        joined = models.db.joinedload
+        base_query = (models.Backup.query.options(joined('assignment'),
+                                                  joined('submitter'),
+                                                  joined('messages'))
+                            .filter(models.Backup.id.in_(subm_keys))
+                            .order_by(models.Backup.created.desc()))
+
+        output = base_query.all()
+        num_subms = len(subm_keys)
+        # output = base_query.limit(limit).offset(offset)
```

Comment:
It looks like the query completes pretty quickly - but sending 1409 backups results in a long page - so I'm going to paginate it. I think it'll require a few changes to the API docs/